### ISSUE TITLE
Bun middleware that allows you to extend `Response`

### ISSUE THREAD

**Issue by EvHaus** [Danilqa/node-file-router#41]
👋 Hey. It's me again.

I'm trying to implement a CORS middleware in Bun that adds some additional headers to all responses. This is proving to be quite difficult at the moment. I see you have [this example](https://github.com/Danilqa/node-file-router/blob/main/examples/bun/middlewares/cors.js) but that's not quite enough to fully handle CORS headers. I'm trying to do something more like [this](https://github.com/oven-sh/bun/issues/5466#issuecomment-1721356533).

I'm wondering if it would be possible for middleware in Bun to be able to extend the `Response` object somehow. At the moment, `await next()` in middleware returns `undefined`. It would be great if it returned the response of the promise chain.

Ideally, I would like to write a middleware like this:

```ts
import type { NextFunction } from 'node-file-router';

const headers = async (req: Request, next: NextFunction) => {
	const res = await next();

	// Ensure all API requests always return these headers
	res.headers.set('Access-Control-Allow-Origin', '*');
	res.headers.set('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');

	return res;
};

export default headers;
```

Do you think this would be possible to add support for?

**Comment by Danilqa** [comment]
@EvHaus Hey! I've released version 0.6.0 with this feature :) Please feel free to reach out if you have any issues or additional suggestions. The documentation can be found [here](https://node-file-router.js.org/docs/middlewares#returning-result), and it aligns with what you described earlier.

**Comment by EvHaus** [comment]
Works beautifully! Thanks again @Danilqa. I really appreciate your fast reponses and updates to the library.
